src/evolutionary_keras/optimizers.py

```diff
@@ -114,18 +105,29 @@ def get_shape(self):
         -------
             `weight_shapes`: a list of the shapes of all trainable weights
         """
-        # Initialize number of nodes
-        self.n_nodes = 0
         # Get trainable weight from the model and their shapes
         trainable_weights = self.model.trainable_weights
         weight_shapes = [weight.shape.as_list() for weight in trainable_weights]
-        # TODO: eventually we should save here a reference to the layer and their
-        # corresponding weights, since the nodes are the output of the layer
-        # and the weights the corresponding to that layer
-        for layer in self.model.layers:
-            self.n_nodes += get_number_nodes(layer)
-        # TODO related to previous TODO: non trianable weights should not be important
-        self.non_training_weights = self.model.non_trainable_weights
+        self.n_nodes = get_number_nodes(self.model)
+
+        # check compatibility of the shape with the NGA optimizer
+        NNshape = None
+        count_nodes = 0
+        for num, layer_shape in enumerate(weight_shapes):
+            num += 1
+            if num % 2 == 0:
+                count_nodes += np.array(layer_shape)
+                if np.array(layer_shape).size != 1:
+                    NNshape = "incompatible"
+            else:
+                if np.array(layer_shape).size != 2:
+                    NNshape = "incompatible"
+        if count_nodes != self.n_nodes:
+            NNshape = "incompatible"
+        if NNshape == "incompatible":
+            raise ValueError(
+                "Using the NGA optimizer requires the network to have a (weight-bias)^n structure, with 1-dimensional bias"
+            )

```

Comment:
Rather than setting having a `NNshape` variable as a string, I would return the exception at every point, defining exactly why the exception happened in each case.